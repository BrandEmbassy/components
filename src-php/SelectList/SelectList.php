<?php declare(strict_types = 1);

namespace BrandEmbassy\Components\SelectList;

use BrandEmbassy\Components\ArrayRenderer;
use BrandEmbassy\Components\UiComponent;

final class SelectList implements UiComponent
{

    /**
     * @var UiComponent[]|string[]
     */
    private $children;

    /**
     * @param UiComponent[]|string[]|UiComponent|string $children
     */
    public function __construct($children)
    {
        $this->children = \is_array($children) ? $children : [$children];
    }

    public function render(): string
    {
        return \sprintf(
            '<div class="select-list-654d">%s</div>',
            ArrayRenderer::render($this->children)
        );
    }

}
