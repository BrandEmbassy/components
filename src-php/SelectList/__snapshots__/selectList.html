<div class="select-list-654d">
    <div class="Checkbox__CheckboxContent" data-reactroot="">
        <div class="Checkbox__Checkbox">
            <input type="checkbox" id="id-123" value="1" name="baz[]" />
            <label for="id-123"></label></div>
        <div class="Checkbox__Label">Foo Bar</div>
    </div>
</div>
